Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/utilities/TimexUtility.java

```diff
@@ -28,33 +30,19 @@
     public static String generateCompoundDurationTimex(Map<String, String> unitToTimexComponents, ImmutableMap<String, Long> unitValueMap) {
         List<String> unitList = new ArrayList<>(unitToTimexComponents.keySet());
         unitList.sort((x, y) -> unitValueMap.get(x) < unitValueMap.get(y) ? 1 : -1);
-        boolean isTimeDurationAlreadyExist = false;
-        StringBuilder timexBuilder = new StringBuilder(Constants.GeneralPeriodPrefix);
-
-        for (String unitKey : unitList) {
-            String timexComponent = unitToTimexComponents.get(unitKey);
-
-            // The Time Duration component occurs first time
-            if (!isTimeDurationAlreadyExist && isTimeDurationTimex(timexComponent)) {
-                timexBuilder.append(Constants.TimeTimexPrefix);
-                timexBuilder.append(getDurationTimexWithoutPrefix(timexComponent));
-                isTimeDurationAlreadyExist = true;
-            } else {
-                timexBuilder.append(getDurationTimexWithoutPrefix(timexComponent));
-            }
+                
+        unitList = new ArrayList<>();
+        for (String unitKey: unitList) {
+        	unitList.add(unitToTimexComponents.get(unitKey));
         }
```

Comment:
Something like:
```suggestion
        unitList = unitList.stream().map(t -> t.get(unitKey));
```

Select is map in Java

Reply:
this isn't working for me. You mean replacing that line inside the for, or entirely replace the for? 
That line alone returns a stream of String, that I need to turn to list with collect(collector.toList)

Reply:
I'll call you

Reply:
This is the correct one `unitList = unitList.stream().map(t -> unitToTimexComponents.get(t)).collect(Collectors.toList());`

Reply:
This is the correct one `unitList = unitList.stream().map(t -> unitToTimexComponents.get(t)).collect(Collectors.toList());`

Reply:
worked with your suggestion 👍